timesketch/lib/analyzers/sequence_sessionizer_test.py

```diff
@@ -0,0 +1,391 @@
+            sessionizer = ManyEventsSequenceSessionizer(index, sketch_id)
+            message = sessionizer.run()
+            self.assertEqual(
+                message,
+                'Sessionizing completed, number of {0} session created: 0'.
+                format(sessionizer.event_seq_name))
+
+            ds = MockDataStore('test', 0)
+            #events with id 0 and id 101 form the requested sequence, but events
+            #with id 100 and 101 have max_time_diff_micros + 1 bewtween them
+            for i in range(0, 201):
+                event = (ds.get_event('test_index', str(i),
+                                      stored_events=True))
+                self.assertNotIn(sessionizer.event_seq_name, event)
+
+    @mock.patch('timesketch.lib.analyzers.interface.ElasticsearchDataStore',
+                MockDataStore)
+    def test_zero_events(self):
+        """Test the behaviour of the sequence sessionizer when given zero
+        events."""
+        with mock.patch.object(SequenceSessionizerSketchPlugin,
+                               'event_stream',
+                               return_value=_create_mock_event(0, 0, [], [0])):
+            index = 'test_index'
+            sketch_id = 1
+            sessionizer = ManyEventsSequenceSessionizer(index, sketch_id)
+            message = sessionizer.run()
+            self.assertEqual(
+                message,
+                'Sessionizing completed, number of {0} session created: 0'.
+                format(sessionizer.event_seq_name))
+
+    @mock.patch('timesketch.lib.analyzers.interface.ElasticsearchDataStore',
+                MockDataStore)
+    def test_one_event_seq(self):
+        """Test the behaviour of the seqeunce sessionizer when given a sequence
+        of one event."""
+        with mock.patch.object(SequenceSessionizerSketchPlugin,
+                               'event_stream',
+                               return_value=_create_mock_event(
+                                   0, 1, one_event_args, [max_allowed])):
+            index = 'test_index'
+            sketch_id = 1
+            sessionizer = OneEventSequenceSessionizer(index, sketch_id)
+            message = sessionizer.run()
+            self.assertEqual(
+                message,
+                'Sessionizing completed, number of {0} session created: 1'.
+                format(sessionizer.event_seq_name))
+
+            ds = MockDataStore('test', 0)
+            event = (ds.get_event('test_index', '0', stored_events=True))
+            self.assertEqual(event['_source'][sessionizer.event_seq_name], 1)
+
+            for i in range(1, 100):
+                event = (ds.get_event('test_index', str(i),
+                                      stored_events=True))
+                self.assertNotIn(sessionizer.event_seq_name, event)
+
+
+def _create_mock_event(event_id, quantity, attributes, time_diffs=None):
+    """
+    Returns an instance of Event, based on the MockDataStore event_dict
```

Comment:
Does this really return an instance of Event? Or rather generates a list of Event objects?